script.js

```diff
@@ -0,0 +1,987 @@
+                this.shift = false;
+            }
+
+            if ((button.dataset.button === 'AltRight') || (button.dataset.button === 'AltLeft')) {
+                this.alt = false;
+            }
+
+            this.changeWordCase()
+
+        }
+    }
+
+    changeWordCase() {
+        let q = document.querySelectorAll(`.${this.lang} span`)
+
+        if (!this.capsLock && !this.shift) {
+            q.forEach((item) => {
+                if (item.classList.contains('noShift')) {
+                    item.classList.remove('hidden');
+                } else if (!item.classList.contains('hidden')) {
+                    item.classList.add('hidden');
+                }
+            });
+
+        } else if (!this.capsLock && this.shift) {
+            q.forEach((item) => {
+                if (item.classList.contains('isShift')) {
+                    item.classList.remove('hidden');
+                } else if (!item.classList.contains('hidden')) {
+                    item.classList.add('hidden');
+                }
+            });
+
+        } else if (this.capsLock && !this.shift) {
+            q.forEach((item) => {
+                if (item.classList.contains('caps')) {
+                    item.classList.remove('hidden');
+                } else if (!item.classList.contains('hidden')) {
+                    item.classList.add('hidden');
+                }
+            });
+        } else if (this.capsLock && this.shift) {
+            q.forEach((item) => {
+                if (item.classList.contains('shiftCaps')) {
+                    item.classList.remove('hidden');
+                } else if (!item.classList.contains('hidden')) {
+                    item.classList.add('hidden');
+                }
+            });
+        }
+    }
+
+    setTextToInput(text) {
+        let startPos = this.textArea.selectionStart;
+        let endPos = this.textArea.selectionEnd;
+
+        this.textArea.value = `${this.textArea.value.substring(0, startPos)}${text}${this.textArea.value.substring(endPos)}`
+
+        this.textArea.focus();
+        this.textArea.selectionStart = startPos + text.length;
+        this.textArea.selectionEnd = startPos + text.length;
+
+    }
+
+    changeLanguage() {
+        let q = document.querySelectorAll('span.rus, span.eng');
+
+        q.forEach((item) => {
+            if (item.classList.contains(this.lang)) {
+                item.classList.remove('hidden')
+            } else {
+                item.classList.add('hidden');
+
+                for (let i = 0; i < item.children.length; i++ ) {
+                    if (!item.children.item(i).classList.contains('hidden')) {
+                        item.children.item(i).classList.add('hidden');
+                    }
+                }
+
+            }
+        });
+
+        this.changeWordCase();
+    }
+}
+
+window.addEventListener('load', () => {
+   console.log('Загружаем клавиатуру');
+   let v_keyboard = new V_keyboard();

```

Comment:
Is this V for Vendetta?

VirtualKeyboard or just Keyboard is a good name

Reply:
done